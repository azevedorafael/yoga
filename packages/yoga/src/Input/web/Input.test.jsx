import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ThemeProvider, Input } from '../..';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with label', () => {
      const { container } = render(
        <ThemeProvider>
          <Input label="Input" />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with disabled input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input disabled />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with error', () => {
      const { container } = render(
        <ThemeProvider>
          <Input error="Error message" />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with helper text and max length', () => {
      const { container } = render(
        <ThemeProvider>
          <Input helper="Helper text" maxLength={20} />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with full width', () => {
      const { container } = render(
        <ThemeProvider>
          <Input label="Label" full />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChange', () => {
      const onChangeMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" data-testid="input" onChange={onChangeMock} />
        </ThemeProvider>,
      );

      fireEvent.change(getByTestId('input'), { target: { value: 'foo' } });

      expect(onChangeMock).toHaveBeenCalled();
    });

    it('should call onFocus', () => {
      const onFocusMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" data-testid="input" onFocus={onFocusMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(onFocusMock).toHaveBeenCalled();
    });

    it('should call onBlur', () => {
      const onBlurMock = jest.fn();

      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" data-testid="input" onBlur={onBlurMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));
      fireEvent.blur(getByTestId('input'));

      expect(onBlurMock).toHaveBeenCalled();
    });
  });

  describe('maxLength', () => {
    it('should update maxLength counter when add character', () => {
      const { getByTestId, getByText } = render(
        <ThemeProvider>
          <Input label="Input" data-testid="input" maxLength={10} />
        </ThemeProvider>,
      );

      expect(getByText('0/10').textContent).toBe('0/10');

      fireEvent.change(getByTestId('input'), { target: { value: 'foo' } });

      expect(getByText('3/10').textContent).toBe('3/10');
    });
  });

  describe('clean button', () => {
    it('should call onClean when press clean button', () => {
      const onCleanMock = jest.fn();

      const { getByRole, getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" data-testid="input" onClean={onCleanMock} />
        </ThemeProvider>,
      );

      fireEvent.change(getByTestId('input'), { target: { value: 'foo' } });
      fireEvent.click(getByRole('button'));

      expect(onCleanMock).toHaveBeenCalledWith('');
    });
  });
});
