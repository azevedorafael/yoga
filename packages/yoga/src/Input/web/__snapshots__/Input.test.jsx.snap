// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match with default input 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c1 {
  width: 320px;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #41414A;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c0 {
  width: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label="Label"
    >
      <input
        class="c2"
        value=""
      />
      <label
        class="c3"
      >
        Label
      </label>
      <fieldset
        class="c4"
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`<Input /> Snapshots should match with disabled input 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
  border-color: #D3D3E2;
  color: #D3D3E2;
}

.c1 {
  width: 320px;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #41414A;
  border-color: #D3D3E2;
}

.c1 svg {
  fill: #D3D3E2;
  pointer-events: none;
}

.c1 fieldset {
  border-color: #D3D3E2;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  color: #D3D3E2;
}

.c0 {
  width: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      disabled=""
      label="Label"
    >
      <input
        class="c2"
        disabled=""
        value=""
      />
      <label
        class="c3"
        disabled=""
      >
        Label
      </label>
      <fieldset
        class="c4"
        disabled=""
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`<Input /> Snapshots should match with error 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
  border-color: #CB3530;
}

.c1 {
  width: 320px;
}

.c1 fieldset {
  border-color: #CB3530;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #CB3530;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #CB3530;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  color: #CB3530;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  color: #9898A7;
  color: #CB3530;
}

.c8 {
  font-size: 12px;
}

.c0 {
  width: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label="Label"
    >
      <input
        class="c2"
        value=""
      />
      <label
        class="c3"
      >
        Label
      </label>
      <fieldset
        class="c4"
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Label
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Error message
      </span>
    </div>
  </div>
</div>
`;

exports[`<Input /> Snapshots should match with full width 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c1 {
  width: 100%;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #41414A;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c0 {
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label="Label"
    >
      <input
        class="c2"
        value=""
      />
      <label
        class="c3"
      >
        Label
      </label>
      <fieldset
        class="c4"
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`<Input /> Snapshots should match with helper text and max length 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c1 {
  width: 320px;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #41414A;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  color: #9898A7;
}

.c8 {
  font-size: 12px;
}

.c9 {
  font-size: 12px;
  margin-left: auto;
}

.c0 {
  width: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label="Label"
    >
      <input
        class="c2"
        maxlength="20"
        value=""
      />
      <label
        class="c3"
      >
        Label
      </label>
      <fieldset
        class="c4"
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Label
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Helper text
      </span>
      <span
        class="c9"
      >
        0
        /
        20
      </span>
    </div>
  </div>
</div>
`;

exports[`<Input /> Snapshots should match with label 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c1 {
  width: 320px;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover fieldset,
.c1:focus-within fieldset {
  border-color: #41414A;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

.c4 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c5 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  padding-left: 4px;
  padding-right: 4px;
}

.c3 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c0 {
  width: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label="Input"
    >
      <input
        class="c2"
        value=""
      />
      <label
        class="c3"
      >
        Input
      </label>
      <fieldset
        class="c4"
      >
        <legend
          class="c5"
        >
          <span
            class="c6"
          >
            Input
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
